pkg/controller/dnszone/dnszone_controller_test.go

```diff
@@ -134,6 +135,14 @@ func TestReconcileDNSProviderForAWS(t *testing.T) {
 			},
 			expectZoneDeleted: true,
 		},
+		{
+			name:    "Delete zone with PreserveOnDelete",
+			dnsZone: validDNSZoneBeingDeletedWithPreserve(),
+			// No AWS calls expected in this case, our creds may be bad:
+			setupAWSMock: func(expect *mock.MockClientMockRecorder) {
```

Comment:
nts: Mock expectations being empty here is the key thing; without PreserveOnDelete we get failures ✓